onnx/shape_inference/implementation.h

```diff
@@ -12,36 +12,101 @@
 namespace ONNX_NAMESPACE {
 namespace shape_inference {
 
+class SymbolTableImpl : public SymbolTable {
+ public:
+  SymbolTableImpl() : index_(0){}
+
+  void addFromGraph(const GraphProto& g) {
+    AddExistingSymbolicDims(g.input());
+    AddExistingSymbolicDims(g.output());
+    AddExistingSymbolicDims(g.value_info());
+  }
+
+  std::string createNew(const std::string& symbol_prefix="unk__") {
```

Comment:
Add comment something like:
Creates a new unique symbol with the given prefix and adds it to the SymbolTable
Returns the newly created symbol